src/wallet/wallet_tx_builder.cpp

```diff
@@ -0,0 +1,683 @@
+        && spendableMut.Total() - spendableMut.GetSproutTotal() >= targetAmount;
+    std::vector<ResolvedPayment> resolvedPayments;
+    std::optional<AddressResolutionError> resolutionError;
+    for (const auto& payment : payments) {
+        std::visit(match {
+            [&](const CKeyID& p2pkh) {
+                if (strategy.AllowRevealedRecipients()) {
+                    resolvedPayments.emplace_back(
+                            std::nullopt, p2pkh, payment.GetAmount(), payment.GetMemo(), false);
+                } else {
+                    resolutionError = AddressResolutionError::TransparentRecipientNotAllowed;
+                }
+            },
+            [&](const CScriptID& p2sh) {
+                if (strategy.AllowRevealedRecipients()) {
+                    resolvedPayments.emplace_back(
+                            std::nullopt, p2sh, payment.GetAmount(), payment.GetMemo(), false);
+                } else {
+                    resolutionError = AddressResolutionError::TransparentRecipientNotAllowed;
+                }
+            },
+            [&](const SproutPaymentAddress&) {
+                resolutionError = AddressResolutionError::SproutRecipientsNotSupported;
+            },
+            [&](const SaplingPaymentAddress& addr) {
+                if (strategy.AllowRevealedAmounts() || payment.GetAmount() < maxSaplingAvailable) {
+                    resolvedPayments.emplace_back(
+                            std::nullopt, addr, payment.GetAmount(), payment.GetMemo(), false);
+                    if (!strategy.AllowRevealedAmounts()) {
+                        maxSaplingAvailable -= payment.GetAmount();
+                    }
+                } else {
+                    resolutionError = AddressResolutionError::RevealingSaplingAmountNotAllowed;
+                }
+            },
+            [&](const UnifiedAddress& ua) {
+                if (canResolveOrchard
+                    && ua.GetOrchardReceiver().has_value()
+                    && (strategy.AllowRevealedAmounts() || payment.GetAmount() < maxOrchardAvailable)
+                    ) {
+                    resolvedPayments.emplace_back(
+                        ua, ua.GetOrchardReceiver().value(), payment.GetAmount(), payment.GetMemo(), false);
+                    if (!strategy.AllowRevealedAmounts()) {
+                        maxOrchardAvailable -= payment.GetAmount();
+                    }
+                    orchardOutputs += 1;
+                } else if (ua.GetSaplingReceiver().has_value()
+                    && (strategy.AllowRevealedAmounts() || payment.GetAmount() < maxSaplingAvailable)
+                    ) {
+                    resolvedPayments.emplace_back(
+                        ua, ua.GetSaplingReceiver().value(), payment.GetAmount(), payment.GetMemo(), false);
+                    if (!strategy.AllowRevealedAmounts()) {
+                        maxSaplingAvailable -= payment.GetAmount();
+                    }
```

Comment:
Similar case to above.

Reply:
See https://github.com/zcash/zcash/pull/6377/files#r1142654730